mathics/eval/arithmetic.py

```diff
@@ -688,8 +694,88 @@ def eval_Times(*items: BaseElement) -> BaseElement:
     )
 
 
+# Here I used the convention of calling eval_* to functions that can produce a new expression, or None
```

Comment:
I don't understand this comment. Based on past experience, I have a sinking feeling this is going off into some misguided direction. (I think of the days when text substitutions on MathML boxes was proposed as a way of addressing formatting?)

Reply:
`eval_` methods are used as the `function` attribute of a `BuiltinRule`. When the rule is applied, the `eval_` method is filled with the parameters from the pattern, and produce an output. If the output is `None`, the rule is assumed to "fail" and the original expression is kept. Otherwise, if the `eval_` method returns an `Element`, then this element is used as the result of the replacement, and the rule is considered  "applied".

Here we have certain `eval_` functions that are not `eval_` methods but can be used as a part of them. If you are up, please help me to rephrase the comment.

Reply:
An alternative way to solve the same problem would be to add a parameter `expression` into the `eval_` methods. Then, we can just check if the result of the replacement is the same as the original expression without any extra cost. However, it will take to change the code in many many places...

Reply:
To simplify this PR, I am going to remove these comments, that still are available in #766.